src/liballoc/boxed.rs

```diff
@@ -395,11 +395,9 @@ impl<T: Clone> Clone for Box<T> {
 #[stable(feature = "box_slice_clone", since = "1.3.0")]
 impl Clone for Box<str> {
     fn clone(&self) -> Self {
-        let len = self.len();
-        let buf = RawVec::with_capacity(len);
+        let buf: Box<[u8]> = self.as_bytes().into();
```

Comment:
```suggestion
        // this makes a copy of the data
        let buf: Box<[u8]> = self.as_bytes().into();
```
I can't help myself, I just love comments. ;)